bridge/client/app/project-board/project-board.component.ts

```diff
@@ -239,6 +226,22 @@ export class ProjectBoardComponent implements OnInit, OnDestroy {
     this.selectedStage = stage;
   }
 
+  countOpenApprovals(openApprovals: Trace[], stage: Stage, service?: Service) {
+    return openApprovals.filter(approval => approval == openApprovals.find(a => a.data.stage == stage.stageName && a.data.service == approval.data.service && (!service || a.data.service == service.serviceName))).length;
+  }
+
+  getOpenApprovals(openApprovals: Trace[], stage: Stage, service: Service) {
+    return openApprovals.filter(approval => approval.data.stage == stage.stageName && approval.data.service == service.serviceName);
+  }
```

Comment:
What do you think of having an  ApprovalService where you can put all the logic of filtering, counting and getting the approvals?

Reply:
Since we plan to nx-ify the bridge #1905, I would address this with that refactoring task.